api_app/classes.py

```diff
@@ -121,9 +123,22 @@ def after_run(self):
         self.report.save()
 
     def after_run_success(self, content: typing.Any):
+        # exhaust generator
         if isinstance(content, typing.Generator):
             content = list(content)
-        self.report.report = content
+        # avoiding JSON serialization errors for types: File and bytes
+        report_content = content
+        if isinstance(report_content, typing.List):
+            if all(isinstance(n, File) for n in report_content):
+                report_content = [
+                    base64.b64encode(f.read()).decode("utf-8") for f in report_content
+                ]
+            elif all(isinstance(n, bytes) for n in report_content):
+                report_content = [
+                    base64.b64encode(b).decode("utf-8") for b in report_content
+                ]
```

Comment:
Since we are iterating them anyway, why don't we convert each object to the correct base64?
```python3
report_content = []
for n in content:
   if isinstance(n, File):
       report_content.append(...)
   elif isinstance(n, bytes):
       report_content.append(...)   
   else: report_content.append(n)   
   ```